app/src/common/shared/com/igalia/wolvic/ui/views/library/DownloadsView.java

```diff
@@ -122,6 +123,22 @@ public void onDestroy() {
         mBinding.downloadsList.removeOnScrollListener(mScrollListener);
     }
 
+    @Override
+    public boolean supportsSearch() {
```

Comment:
This repetition of the same pattern in all the views makes me wonder why we don't move everything to the LibraryView class.

Reply:
The reason for `supportsSearch()` is that not all of the views support searching at the moment. I'm not sure that it would be useful in all cases; for example, we currently only list 5 add-ons.

We can not move the code that filters the list to the superclass, because it is different for each view (the types of items and adapters are different, etc.).

I did have some doubts about putting this other bit in the superclass, but I did it in the end because it was an easy way to enforce that non-null constraint:

```java
protected void setSearchFilter(String searchFilter) {
...
@NonNull
protected String getSearchFilter()
```

Reply:
OK fair enough